omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -951,6 +963,13 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
 
         _doSelected: function(position, rowData) {
             if(this.options.selectionHandler) {
+		// Check the browser compatibility and, add an entry to history
+		if (window.history.pushState) {
+			window.history.pushState(
+				{}, "",
+				document.location.pathname + "?lastSearchText=" + document.getElementById('inputNode').value
```

Comment:
You should not set lastSearchText as a request parameter, it should be in the state object you push into the history, same for includeVoided and showVerbose